### PR TITLE
[DIOPI]upgrade flash_attn to 2.4.3

### PR DESCRIPTION
**Pull request by ustclight-sls**
upgrade flash_attn to 2.4.3

### CODE REVIEW

File: impl/torch/functions/functions_ext.cpp [hunk 1]
```diff
@@ -236,15 +243,15 @@ diopiError_t diopiFlashAttention(diopiContextHandle_t ctx, diopiTensorHandle_t a
                                  float p_dropout, float softmax_scale, bool is_causal, int32_t window_size_left, int32_t window_size_right) {
     impl::aten::setCurStream(ctx);
 
-    // handle param[out]
+    // handle param[out]mha_bwd
```
Comment (lljbash) [inline impl/torch/functions/functions_ext.cpp:246]:
```suggestion
    // handle param[out]
```

File: impl/torch/functions/functions_ext.cpp [hunk 2]
```diff
@@ -236,15 +243,15 @@ diopiError_t diopiFlashAttention(diopiContextHandle_t ctx, diopiTensorHandle_t a
...
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optOut, attention_out);
+    DIOPI_IMPL_BUILD_ATEN_OPTIONAL(linearBiasSlopes, alibi_slopes);
 
     // handle param[in]
     auto atGen = buildGeneratorForMha(ctx, gen, p_dropout);
     auto atQ = impl::aten::buildATen(q);
     auto atK = impl::aten::buildATen(k);
     auto atV = impl::aten::buildATen(v);
```
Comment (lljbash) [inline impl/torch/functions/functions_ext.cpp:254]: 这是个 in param，还有命名一致性
```suggestion

    // handle param[in]
    auto atGen = buildGeneratorForMha(ctx, gen, p_dropout);
    auto atQ = impl::aten::buildATen(q);
    auto atK = impl::aten::buildATen(k);
    auto atV = impl::aten::buildATen(v);
    DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optLinearBiasSlopes, alibi_slopes);
```

File: impl/torch/functions/functions_ext.cpp [hunk 3]
```diff
@@ -279,14 +287,14 @@ diopiError_t diopiFlashAttentionBackward(diopiContextHandle_t ctx, diopiTensorHa
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradQ, grad_q);
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradK, grad_k);
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradV, grad_v);
+    DIOPI_IMPL_BUILD_ATEN_OPTIONAL(linearBiasSlopes, alibi_slopes);
```
Comment (lljbash) [inline impl/torch/functions/functions_ext.cpp:290]: ditto

File: impl/torch/functions/functions_ext.cpp [hunk 4]
```diff
@@ -320,6 +330,7 @@ diopiError_t diopiFlashAttentionVarLen(diopiContextHandle_t ctx, diopiTensorHand
 
     // handle param[out]
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optOut, attention_out);
+    DIOPI_IMPL_BUILD_ATEN_OPTIONAL(linearBiasSlopes, alibi_slopes);
```
Comment (lljbash) [inline impl/torch/functions/functions_ext.cpp:333]: ditto

File: impl/torch/functions/functions_ext.cpp [hunk 5]
```diff
@@ -370,6 +384,7 @@ diopiError_t diopiFlashAttentionVarLenBackward(diopiContextHandle_t ctx, diopiTe
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradQ, grad_q);
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradK, grad_k);
     DIOPI_IMPL_BUILD_ATEN_OPTIONAL(optGradV, grad_v);
+    DIOPI_IMPL_BUILD_ATEN_OPTIONAL(linearBiasSlopes, alibi_slopes);
```
Comment (lljbash) [inline impl/torch/functions/functions_ext.cpp:387]: ditto

**Review by lljbash — APPROVED** [review]

**Review by POI-WX — APPROVED** [review]

**Review by yangbofun — APPROVED** [review]

**Pull request merged**